api/desecapi/permissions.py

```diff
@@ -121,6 +112,39 @@ def has_permission(self, request, view):
         return ip in IPv4Network("10.8.0.0/24")
 
 
+class HasCreateDomainPermission(permissions.BasePermission):
+    """
+    Permission to check whether a token has "create domain" permission.
+    """
+
+    def has_permission(self, request, view):
+        return request.auth.perm_create_domain
+
+
+class HasDeleteDomainPermission(permissions.BasePermission):
+    """
+    Permission to check whether a token has "delete domian" permission.
+    """
+
+    def has_permission(self, request, view):
+        return request.auth.perm_delete_domain
+
+    def has_object_permission(self, request, view, obj):
+        policy_set = request.auth.tokendomainpolicy_set
+
+        return (
+            # Ensure token is not explicitly prohibited from writing some RRsets in this domain
+            not policy_set.filter(domain=obj, perm_write=False).exists()
+            and
+            # Ensure applicable domain-independent policies are not constraining
```

Comment:
stale comment, should be "applicable default policies"

Reply:
done